src/input/target/steam_deck.rs

```diff
@@ -718,6 +765,77 @@ impl SteamDeckDevice {
 }
 
 impl TargetInputDevice for SteamDeckDevice {
+    /// Start the driver when attached to a composite device.
+    fn on_composite_device_attached(
+        &mut self,
+        composite_device: CompositeDeviceClient,
+    ) -> Result<(), InputError> {
+        let (tx, rx) = channel(1);
+        let mut device_config = self.config.clone();
+
+        // Spawn a task to wait for the composite device config. This is done
+        // to prevent potential deadlocks if the composite device and target
+        // device are both waiting for a response from each other.
+        tokio::task::spawn(async move {
+            // Wait to ensure the composite device has grabbed all sources
+            // NOTE: We should look at other ways of signalling to target devices
+            // that a new source device has been added.
+            tokio::time::sleep(Duration::from_millis(500)).await;
```

Comment:
Since you are just requesting the config from the composite device and not querying the source devices, we shouldn't need this here:

```suggestion
```